### PR TITLE
Utilisation rate

### PR DESCRIPTION
**Pull request by bachstatter**

### CODE REVIEW

**Review by davidvuong — COMMENTED** [review]
Got a few more comments but need some more time to read through the dense areas of utilisation calc

File: contracts/interfaces/IPerpAccountModule.sol [hunk 1]
```diff
@@ -44,6 +44,8 @@ interface IPerpAccountModule {
         uint256 accruedFeesUsd;
         // Funding accrued in USD.
```
[the remark this answers is not in the document]
Comment (bachstatter) [reply contracts/interfaces/IPerpAccountModule.sol:45]: This is coming from health data, which calls `getUnrecordedFunding` which is taking current price into consideration --> making this a USD value
```
healthData.accruedFunding = positionSize.mulDecimal(
   market.getUnrecordedFunding(price) + market.currentFundingAccruedComputed - positionEntryFundingAccrued
);
```

[the remark this answers is not in the document]
Comment (bachstatter) [reply contracts/interfaces/IPerpAccountModule.sol:45]: the unrecorded funding, which is in USD is also what is updating `currentFundingAccruedComputed`. Which mean that's the comment that is wrong https://github.com/Synthetixio/bfp-market/pull/51/commits/f510c376e0294e1cd5b79f09ebb24a2c4811b7e2

File: contracts/storage/PerpMarket.sol [hunk 2]
```diff
@@ -240,7 +311,12 @@ library PerpMarket {
             (DecimalMath.UNIT * 2).toInt()
         );
         // Calculate the additive accrued funding delta for the next funding accrued value.
-        unrecordedFunding = avgFundingRate.mulDecimal(getProportionalElapsed(self)).mulDecimal(price.toInt());
+        unrecordedFunding = avgFundingRate.mulDecimal(getProportionalFundingElapsed(self)).mulDecimal(price.toInt());
+    }
+
+    function getUnrecordedFunding(PerpMarket.Data storage self, uint256 price) internal view returns (int256) {
```
Comment (bachstatter) [inline contracts/storage/PerpMarket.sol:317]: See if we can just destructure at the call site

**Review by davidvuong — APPROVED** [review]
Approving now to unblock but still looking for those changes we discussed on DC before we merge.

**Pull request merged**